### PR TITLE
feat(ui): Add statistics button

### PR DESCRIPTION
**Pull request by nbcl**
The present pull request provides a solution to #2607 (Add stats for nerds to UI (in the same way as YouTube)) by adding a new statistics button to the UI.

## Example

<p align=center>

</p>

## How to preview

Initialize the client with a `config` that includes `statistics` on either the `controlPanelElements` or `overflowMenuButtons` sections and optionally, select the desired `statisticsList` to display, as follows.

```js
const config = {
    'controlPanelElements': [... 'statistics', ...],
    'statisticsList': [... 'playTime', 'pauseTime', 'width', 'height', ...]
};
ui.configure(config);
``` 

## Changelog

<details>
  <summary><code>build/types/ui</code></summary>

Added the `statistics_button.js` path to the UI types file. 
</details>

<details>
  <summary><code>docs/tutorials/ui-customization.md</code></summary>

The `statistics` button was added to both `controlPanelElements` and `overflowMenuButtons` lists, and an aditional section was added to explain the list configuration.
</details>

<details>
  <summary><code>shaka-player.uncompiled.js</code></summary>

Adds `shaka.ui.StatisticsButton` for the uncompiled version of the library.
</details>

<details>
  <summary><code>test/ui/ui_unit.js</code></summary>

Adds tests to the button. On the second test, although there is a delay of 0.2, only a buffering time of 0.1 is expected (`expect(bufferingTime).toBeGreaterThan(0.1);`), this is due to the `shaka.util.Timer` on the button only ticking every 0.1.
</details>

<details>
  <summary><code>ui/enums.js</code></summary>

Adds two additional Material Design Icons to be used in the statistics button.
</details>

<details>
  <summary><code>ui/externs/ui.js</code></summary>

The new `statisticsList` is added, giving the option to select specific statistics for the button.
</details>

<details>
  <summary><code>ui/less/container.less</code></summary>

A new `.shaka-statistics-container` class is created to declare the styles of the container.
</details>

<details>
  <summary><code>ui/locales/en.json</code></summary>

The statistics button name is added to `en.json` as an example. 
</details>

<details>
  <summary><code>ui/locales/source.json</code></summary>

The statistics button name and description are added to the `locales` source. 
</details>

<details>
  <summary><code>ui/statistics_button.js</code></summary>

Using `shaka.util.Timer`, `player.get_stats()` is called periodically when the statistics container is displayed.
</details>

<details>
  <summary><code>ui/ui.js</code></summary>

The default configuration of `statisticsList` is added, displaying almost all of the available statistics, with the exception of `stateHistory` and `switchHistory`. Additionally, they are displayed in alphabetical order with the exception of `width` and `height`, which were moved to the top.
</details>

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made the corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have run ./build/all.py and the build passes
- [x] I have run ./build/test.py and all tests pass

### CODE REVIEW

**Comment by michellezhuogg** [comment]
Looks great, thank you! 
An enhancement question: is it possible to show the "statistics" menu on right click instead of in the overflow menu / control menu, similar to Youtube?
Screen Shot 2021-07-13 at 8 24 12 AM

**Comment by nbcl** [comment]
Thank you Shaka Bot, this is an interesting edge case.

Due to the timer ticking every `0.1`, I allowed the test to have a margin by assigning `0.2` seconds for the delay, but it appears to be that if run "fast" enough, `getStatsFromContainer()` is called before the container is updated by the timer for a second time. I ran the test multiple times locally, and rightfully so, it eventually failed.

Fixed it on my last commit!

File: docs/tutorials/ui-customization.md [hunk 1]
```diff
@@ -104,6 +106,19 @@ customized with `addSeekBar` and `addBigPlayButton` booleans in the config.
 UI layout can be reconfigured at any point after it's been created.
 Please note that custom layouts might need CSS adjustments to look good.
 
+#### Configuring Statistics
+
+The list of statistics that are displayed when toggling the statistics button can be customized by specifing a `statisticsList` on the configuration. By default, all of the following are activated.
```
Comment (theodab) [inline docs/tutorials/ui-customization.md:111]: It'd be helpful to link to the `Stats` extern, which is what this is displaying.
E.g.: `This can display any value from the player stats. See {@link shaka.extern.Stats here} for a full list of options.`

**Review by theodab — CHANGES_REQUESTED** [review]

File: docs/tutorials/ui-customization.md (hunk 1 above)
Comment (nbcl) [reply docs/tutorials/ui-customization.md:111]: Thank you Theodab, I linked the extern on the refactor!

**Comment by nbcl** [comment]
Thank you Michelle and Theodab for the feedback - I refactored my approach to include a configurable right-click context menu and a new visual interface for the container on #3548!

**Comment by ManuelBradicic42** [comment]
Hi,
I put the snippet of code in the myapp.js in init() function. 
The general error I get is 'Cannot read properties of undefined (reading 'configure')'

**Comment by ManuelBradicic42** [comment]
Heya, thanks I managed to set it up. I was wondering if it is possible to add some other custom variables into statistics. I believe it is JSON object of player.getStats(), but is it possible to forward the object which I previously edited.
The reason is for instance, the frame rate can be deduced from decoded frames and I would like to add some other dynamic strings which would give me better overview.
